kernel/syscall.c

```diff
@@ -2933,6 +2933,30 @@ long myst_syscall_prlimit64(
             old_rlim->rlim_max = MYST_PROCESS_MAX_STACK_SIZE;
         }
     }
+    else if (resource == RLIMIT_NPROC)
+    {
+        if (old_rlim)
+        {
+            old_rlim->rlim_cur = __myst_kernel_args.max_threads;
+            old_rlim->rlim_max = __myst_kernel_args.max_threads;
+        }
+    }
+    else if (resource == RLIMIT_AS)
```

Comment:
The cases for RLIMIT_AS and RLIMIT_FSIZE can probably be folded together.

Reply:
I would prefer to keep them as separate cases as we look into supporting setting RLIMIT.